### PR TITLE
fix: Improve ZAPI template selection

### PR DESCRIPTION
**Pull request by rahulguptajss**
fixes#322

### CODE REVIEW

File: cmd/poller/collector/helpers.go [hunk 1]
```diff
@@ -103,10 +102,26 @@ func (c *AbstractCollector) ImportSubTemplate(model, filename string, version [3
 	}
 
 	subTemplateFp = path.Join(pathPrefix, selectedVersion, filename)
-	c.Logger.Debug().Msgf("selected best-fitting subtemplate [%s]", subTemplateFp)
+	c.Logger.Info().Msgf("selected best-fitting subtemplate [%s]", subTemplateFp)
```
Comment (Hardikl) [inline cmd/poller/collector/helpers.go:105]: intended?

**Review by Hardikl — DISMISSED** [review]

File: cmd/poller/collector/helpers.go (hunk 1 above)
Comment (rahulguptajss) [reply cmd/poller/collector/helpers.go]: was for testing during draft PR. removed

**Comment by rahulguptajss** [comment]
edge cases covered. Thanks @cgrinds for the code improvements.

**Review by cgrinds — COMMENTED** [review]
Looks good @rahulguptajss
three minor suggestion

File: cmd/poller/collector/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"github.com/hashicorp/go-version"
+	"sort"
+	"testing"
+)
+
+func setupVersions(availableVersions []string) []*version.Version {
+	versions := make([]*version.Version, len(availableVersions))
+	for i, raw := range availableVersions {
+		v, _ := version.NewVersion(raw)
```
Comment (cgrinds) [inline cmd/poller/collector/helpers_test.go:12]: minor: could simply these two lines to `buildVersion(raw)`

File: cmd/poller/collector/helpers.go [hunk 3]
```diff
@@ -81,20 +80,20 @@ func (c *AbstractCollector) ImportSubTemplate(model, filename string, version [3
...
-	for max := 0; max <= 100; max++ {
-		// check older version
-		str := strings.Join(strings.Split(strconv.Itoa(versionDecimal-max), ""), ".")
-		if _, exists := availableVersions[str]; exists == true {
-			selectedVersion = str
-			break
+	if len(availableVersions) > 0 {
+		versions := make([]*version.Version, len(availableVersions))
+		for i, raw := range availableVersions {
+			v, _ := version.NewVersion(raw)
```
Comment (cgrinds) [inline cmd/poller/collector/helpers.go:86]: while it's unlikely seems better to handle err here

File: cmd/poller/collector/helpers.go [hunk 4]
```diff
@@ -81,20 +80,20 @@ func (c *AbstractCollector) ImportSubTemplate(model, filename string, version [3
...
 		}
-		// check newer version
-		str = strings.Join(strings.Split(strconv.Itoa(versionDecimal+max), ""), ".")
-		if _, exists := availableVersions[str]; exists == true {
-			selectedVersion = str
-			break
+
+		sort.Sort(version.Collection(versions))
+
+		verS, _ := version.NewVersion(strings.Trim(strings.Join(strings.Fields(fmt.Sprint(ver)), "."), "[]"))
```
Comment (cgrinds) [inline cmd/poller/collector/helpers.go:92]: while it's unlikely seems better to handle err here

File: cmd/poller/collector/helpers_test.go (hunk 2 above)
Comment (cgrinds) [reply cmd/poller/collector/helpers_test.go:12]: done

File: cmd/poller/collector/helpers.go (hunk 3 above)
Comment (cgrinds) [reply cmd/poller/collector/helpers.go:86]: done

**Comment by cgrinds** [comment]
applied the review comments and validated with `9.10.1`

**Review by cgrinds — APPROVED** [review]

**Pull request merged**